Marlin/src/gcode/calibrate/G34_M422.cpp

```diff
@@ -144,6 +152,11 @@ void GcodeSuite::G34() {
          SQRT(_MAX(HYPOT2(z_stepper_align.xy[0].x - z_stepper_align.xy[1].x, z_stepper_align.xy[0].y - z_stepper_align.xy[1].y),
                    HYPOT2(z_stepper_align.xy[1].x - z_stepper_align.xy[2].x, z_stepper_align.xy[1].y - z_stepper_align.xy[2].y),
                    HYPOT2(z_stepper_align.xy[2].x - z_stepper_align.xy[0].x, z_stepper_align.xy[2].y - z_stepper_align.xy[0].y)))
+      #elif NUM_Z_STEPPER_DRIVERS == 4
+         SQRT(_MAX(HYPOT2(z_stepper_align.xy[0].x - z_stepper_align.xy[1].x, z_stepper_align.xy[0].y - z_stepper_align.xy[1].y),
+                   HYPOT2(z_stepper_align.xy[1].x - z_stepper_align.xy[2].x, z_stepper_align.xy[1].y - z_stepper_align.xy[2].y),
+                   HYPOT2(z_stepper_align.xy[2].x - z_stepper_align.xy[3].x, z_stepper_align.xy[2].y - z_stepper_align.xy[3].y),
+                   HYPOT2(z_stepper_align.xy[3].x - z_stepper_align.xy[0].x, z_stepper_align.xy[3].y - z_stepper_align.xy[0].y)))
```

Comment:
I think you also need compares between 0/2 and 1/3.

If your motors are numbered sequentially around the bed, this will detect the longest edge, but will not detect the diagonal distance between the corner motors.

Personally, I would rather just see a loop that works for any number of motors. (untested)

```
float max_distance = 0.0;
LOOP_LE_N(z1, NUM_Z_STEPPER_DRIVERS)
  LOOP_S_LE_N(z2, z1 + 1, NUM_Z_STEPPER_DRIVERS) {
    auto delta = z_stepper_align.xy[z1] - z_stepper_align.xy[z2];
    max_distance = max(max_distance, HYPOT(delta.x, delta.y));
  }
```

Reply:
A loop did come to mind but then I didnt want to push too many changes with limited ability to test as I dont have all my quad z hardware in hand yet. Im inclined to leave it as-is for the moment and go to a loop when I have more in hand to validate with.

Reply:
You could easily add the two missing checks individually, without going to the loop.